File: python/백준/집합과 맵/25192 인사성 밝은 곰곰이.py
import sys
input = sys.stdin.readline

N = int(input())
count = 0
_seen = set()  # set 사용이 더 간단하고 빠름

for _ in range(N):
    nickname = input().strip()
    
    if nickname == "ENTER":
        _seen.clear()  # set 초기화
    elif nickname not in _seen:
        _seen.add(nickname)
        count += 1

print(count)

File: cpp/백준/집합과 맵/25192 인사성 밝은 곰곰이.cpp
#include <iostream>
#include <string>
#include <unordered_set>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;
    int count = 0;
    unordered_set<string> _seen;

    for (int i = 0; i < N; ++i) {
        string nickname;
        cin >> nickname;
        if (nickname == "ENTER") {
            _seen.clear();
        } else if (_seen.find(nickname) == _seen.end()) {
            _seen.insert(nickname);
            count += 1;
        }
    }

    cout << count << "\n";
    return 0;
}
